Core/Graphics/Include/Babylon/Graphics.h

```diff
@@ -31,6 +31,8 @@ namespace Babylon
             FinishRenderingCurrentFrame();
         }
 
+        void SetDiagnosticOutput(std::function<void(const char* output)> outputFunction);
```

Comment:
I don't think it's a huge deal, but contract-wise this feels a little weird. While it's not exposing anything substantially new through this contract *per se*, using this method (unless I misunderstand it's purpose) will sign the user up to receive diagnostic messages from bgfx, which from the perspective of this contract is an implementation detail. A little odd, but maybe not *too* odd, so I can think of three ways we could address this.
1. Leave it here as-is and just document that its behavior exposes diagnostic information that is beyond the scope of our own contract.
2. Have this function only on the `Impl` such that it is part of the internal contract, where the exposure of implementation details and the expectation that users know what they're doing are not new.
3. Leave this here but find some way to designate it as something tricky, perhaps by something as simple as renaming it or by putting it on an internal object or something.

I personally kind of like option 2, but I think any of them could work. @bghgary, thoughts or preferences?

Reply:
If I move this function to the impl, how can I call it to set the output in the validation tests app?

Reply:
Option 2 doesn't seem possible. This is exposed all the way to the app level. I'm not sure we should expose this on the graphics level though. I would think if we want some diagnostic output, then it should be global to all the modules we write and not just graphics. Hooking up bgfx debug output is just part of that story which probably should be configurable.